sdk/azidentity/uuid.go

```diff
@@ -21,15 +21,18 @@ const (
 type uuid [16]byte
 
 // NewUUID returns a new uuid using RFC 4122 algorithm.
-func newUUID() (u uuid) {
-	u = uuid{}
+func newUUID() (uuid, error) {
+	u := uuid{}
 	// Set all bits to randomly (or pseudo-randomly) chosen values.
-	rand.Read(u[:])
+	_, err := rand.Read(u[:])
```

Comment:
`crypt/rand/Read()` is not no-fail (unlike `math/rand/Read()`) which is why I made this change.  Do we actually need the CRNG here?

Reply:
@JeffreyRichter there are two `uuid.go` files.  The one in `azidentity` is used when creating JWTs and is the one that uses the CRNG.  I don't know if this was by accident or a requirement.  The `uuid.go` in `azcore` is using `math/rand`.